addons/html_editor/static/src/main/link/link_popover.js

```diff
@@ -63,12 +77,99 @@ export class LinkPopover extends Component {
+
+        if (this.props.allowCustomStyle) {
+            this.customTextColorPicker = useColorPicker(
+                "customTextColorButton",
+                {
+                    state: this.customTextColorState,
+                    getUsedCustomColors: () => [],
+                    colorPrefix: "text-",
+                    applyColor: (colorValue) => {
+                        this.customTextColorState.selectedColor = colorValue;
+                        this.customTextResetPreviewColor = colorValue;
+                    },
+                    applyColorPreview: (colorValue) => {
+                        this.customTextColorState.selectedColor = colorValue;
+                    },
+                    applyColorResetPreview: () => {
+                        this.customTextColorState.selectedColor = this.customTextResetPreviewColor;
+                    },
+                },
+                {
+                    onClose: this.onChange.bind(this),
+                }
+            );
+            this.customFillColorPicker = useColorPicker(
+                "customFillColorButton",
+                {
+                    state: this.customFillColorState,
+                    getUsedCustomColors: () => [],
+                    colorPrefix: "bg-",
+                    applyColor: (colorValue) => {
+                        this.customFillColorState.selectedColor = colorValue;
+                        this.customFillResetPreviewColor = colorValue;
+                    },
+                    applyColorPreview: (colorValue) => {
+                        this.customFillColorState.selectedColor = colorValue;
+                    },
+                    applyColorResetPreview: () => {
+                        this.customFillColorState.selectedColor = this.customFillResetPreviewColor;
+                    },
+                },
+                {
+                    onClose: this.onChange.bind(this),
+                }
+            );
+            this.customBorderColorPicker = useColorPicker(
+                "customBorderColorButton",
+                {
+                    state: this.customBorderColorState,
+                    getUsedCustomColors: () => [],
+                    colorPrefix: "border-",
+                    applyColor: (colorValue) => {
+                        this.customBorderColorState.selectedColor = colorValue;
+                        this.customBorderResetPreviewColor = colorValue;
+                    },
+                    applyColorPreview: (colorValue) => {
+                        this.customBorderColorState.selectedColor = colorValue;
+                    },
+                    applyColorResetPreview: () => {
+                        this.customBorderColorState.selectedColor =
+                            this.customBorderResetPreviewColor;
+                    },
+                },
+                {
+                    onClose: this.onChange.bind(this),
+                }
```

Comment:
Consider avoiding code repetition with a helper function `getProps` or something, parametrized for `state` and `colorPrefix`.
The state objects defined above could have a `resetPreviewColor` key to make this easier, and we would no longer need ` this.customTextResetPreviewColor`, `this.customFillResetPreviewColor` and `this.customBorderResetPreviewColor`.